docs/developer/forms-and-validation.md

```diff
@@ -49,17 +49,47 @@ If a form element was repeated for every row in a table, it would make the UI sl
+2. If the validation is passed, the function returns `undefined`.
+3. If the validation fails, the function returns a string to be rendered by the field component as an error.
+
+In addition to the "rules" parameter existing on the LabeledInput and LabeledSelect form components, an additional component called Error (found at `shell/components/form/Error`) exists which can also accept a value and rules. This allows the developer to display an error message which may not be tied to a specific field or tied to a field that is somehow incapable of displaying it's own validation message.
+
+Should a field displaying a validation message be hidden from the user due to form navigation elements like a non-active tab, the developer should still take care to communicate that a validation error exists. To that end, the `Tab` component accepts a simple parameter called "Error" which is simply a boolean value. When true, the "Error" parameter on the `Tab` component will display an error icon provided the tab is not currently active (if the tab is active, the field in question should provide feedback to the user).
+
+## Form Validation
+
+While individual fields are typically responsible for displaying their own validation messages, the form itself should also contain logic to disable the save button should validation errors exist, display errors not bound to a specific field, and/or display errors returned by the API. This is mostly accomplished via a mixin called `form-validation` (found at `shell/mixins/form-validation.js`). Eventually this mixin will be embedded into the existing `CreateEditView` mixin already used by every form but is currently seperate to facilitate a safer implementation of the new Form Validation logic.
+
+To use the `form-validation` mixin, import the mixin and include it in `mixins` in the component like any other mixin. Once included, ensure the following:
+1. Set the "validation-passed" property on the `CruResource` component to "fvFormIsValid" (computed property provided by the mixin). This conditionally disables the "save" button on the form.
+2. Set the "errors" property on the `CruResource` component to "fvUnreportedValidationErrors" (computed property provided by the mixin) or some other value that aggregates errors not otherwise shown in the form as a fallback means of displaying error state to the user.
+
+The `form-validation` mixin itself includes most of the information a developer will need to use it in comments in the file itself but a high-level summary would cover the following points:
+
+1. The specific rules to be validated by the form should be located in a data property for the component named `fvFormRuleSets` which is an array of objects defining the specific rulesets for specific paths in the forms value to validate. The specific properties of the objects required by this array are defined in the mixin file itself as comments.
```

Comment:
I think it would help to have a minimalistic code example below these steps. Basically the simplest possible Vue component that uses this mixin